src/main/java/com/googlecode/jsendnsca/encryption/TripleDESEncryptor.java

```diff
@@ -39,19 +39,19 @@
      * byte[], java.lang.String)
      */
     public void encrypt(byte[] passiveCheckBytes, byte[] initVector, String password) {
-        final byte[] keyBytes = toFixedSizeByteArray(password.getBytes(), 24);
-        final byte[] initVectorBytes = toFixedSizeByteArray(initVector, 8);
+        try {
+            final byte[] keyBytes = toFixedSizeByteArray(password.getBytes("US-ASCII"), 24);
+            final byte[] initVectorBytes = toFixedSizeByteArray(initVector, 8);
 
-        final SecretKey key = new SecretKeySpec(keyBytes, DES_ALGORITHM);
-        final IvParameterSpec iv = new IvParameterSpec(initVectorBytes);
+            final SecretKey key = new SecretKeySpec(keyBytes, DES_ALGORITHM);
+            final IvParameterSpec iv = new IvParameterSpec(initVectorBytes);
 
-        try {
             final Cipher cipher = Cipher.getInstance(DES_TRANSFORMATION);
             cipher.init(ENCRYPT_MODE, key, iv);
             final byte[] cipherText = cipher.doFinal(passiveCheckBytes);
 
             System.arraycopy(cipherText, 0, passiveCheckBytes, 0, passiveCheckBytes.length);
-        } catch (GeneralSecurityException e) {
+        } catch (Exception e) {
```

Comment:
Whats the reason for weakening from GeneralSecurityException to Exception?

Reply:
charset is correctly handled in BlowfishEncryptor and AESEncryptor and they catch Exception.
To be consistent, i've used the same pattern.